src/full_node/blockchain.py

```
@@ -147,7 +149,21 @@ async def _load_chain_from_store(self,) -> None:
             heights = [b.height for b in cur]
             i = heights.index(max(heights))
             self.headers[cur[i].header_hash] = cur[i]
-            cur[i] = headers_db[cur[i].prev_header_hash]
+            prev: Header = headers_db[cur[i].prev_header_hash]
+
+            try:
+                challenge_hash = self.block_store.get_challenge_hash(
+                    cur[i].header_hash
+                )
+                self.block_store.add_proof_of_time(
+                    challenge_hash,
+                    uint64(cur[i].data.total_iters - prev.data.total_iters),
+                    cur[i].data.height,
+                )
+            except Exception as e:
+                log.error(f"Database corruption: {type(e)} {e}")
```

We should probably let this crash as opposed to catching it.